lib/dolly/document.rb

```diff
@@ -1,9 +1,13 @@
 require "dolly/query"
 require "dolly/property"
+require 'dolly/timestamps'
+require 'active_model/callbacks'
 
 module Dolly
   class Document
     extend Dolly::Connection
+    extend ActiveModel::Callbacks
```

Comment:
@neojin We are noticing that in our application logic (in more than 1 app, all of them that use dolly in fact) that we have a legitimate use for timestamps on documents.  My proposal is to abstract this to the gem layer so it doesn't have to be included in every model that inherits from `Dolly::Document`.  This is just on implementation, and Im all ears on if there are better implementations to have timestamps without including `ActiveModel::Callbacks`